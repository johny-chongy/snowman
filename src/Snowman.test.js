import React from 'react'
import Snowman from './Snowman'
import { render, fireEvent } from '@testing-library/react'

////smoketest
test('does not crash on load', function () {
  render(<Snowman words='apple' />)
})

test('Snapshot test', function () {
  const { container, debug } = render(<Snowman words='apple' />)
  const badGuesses = ['b', 'c', 'd', 'f', 'g', 'r']
  const badButtons = badGuesses.map(char =>
    container.querySelector(`button[value="${char}"]`)
  )

  for (let guess of badButtons) {
    fireEvent.click(guess)
  }
  debug(container)
  expect(container).toMatchSnapshot()
})
