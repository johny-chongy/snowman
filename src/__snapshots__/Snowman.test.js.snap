// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
       Wrong guesses: 
      6
       
    </p>
    <p
      class="Snowman-word"
    >
      _
    </p>
    <p>
      You lose
    </p>
  </div>
</div>
`;
